testplan/web_ui/testing/src/Common/ErrorBoundary.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+class ErrorBoundary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { error: null };
+  };
+
+  static getDerivedStateFromError(error) {
+    return { error: error };
+  }
+
+  componentDidCatch(error, info) {
+    this.logErrorToServices(error, info.componentStack);
+  }
```

Comment:
Given I am not an expert here: basically the getDerivedStateFromError and componentDidCatch methods ensure the capture of the error updates the error field of the state, correct? I was looking at this for getting a sense https://legacy.reactjs.org/docs/error-boundaries.html but did not dig into it much.

Reply:
> If you define static getDerivedStateFromError, React will call it when a child component (including distant children) throws an error during rendering. This lets you display an error message instead of clearing the UI.
> 
> Typically, it is used together with [componentDidCatch](https://react.dev/reference/react/Component#componentDidCatch) which lets you send the error report to some analytics service. A component with these methods is called an error boundary.
Source: https://react.dev/reference/react/Component#static-getderivedstatefromerror